pkg/utils/release/controller.go

```diff
@@ -109,29 +114,146 @@ func (s *SuiteController) GetPipelineRunInNamespace(namespace, releaseName, rele
+	var releasePlan *appstudiov1alpha1.ReleasePlan
+
+	if AutoReleaseLabel != "" { // AutoReleaseLabel in ReleasePlan is not missing
+		releasePlan = &appstudiov1alpha1.ReleasePlan{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: name,
+				Name:         name,
+				Namespace:    source_namespace,
+				Labels: map[string]string{
+					appstudiov1alpha1.AutoReleaseLabel: AutoReleaseLabel,
+				},
+			},
+			Spec: appstudiov1alpha1.ReleasePlanSpec{
+				DisplayName: name,
+				Application: application,
+				Target: kcp.NamespaceReference{
+					Namespace: target_namespace,
+				},
+			},
+		}
+	} else { // AutoReleaseLabel in ReleasePlanAdmission is missing
+		releasePlan = &appstudiov1alpha1.ReleasePlan{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: name,
+				Name:         name,
+				Namespace:    source_namespace,
+				Labels:       map[string]string{},
+			},
+			Spec: appstudiov1alpha1.ReleasePlanSpec{
+				DisplayName: name,
+				Application: application,
+				Target: kcp.NamespaceReference{
+					Namespace: target_namespace,
+				},
+			},
+		}
+	}
+	return releasePlan, s.KubeRest().Create(context.TODO(), releasePlan)
+}
+
+// GetReleasePlan returns the releasePlan with the given name in the given namespace.
+func (s *SuiteController) GetReleasePlan(releasePlanName, devNamespace string) (*appstudiov1alpha1.ReleasePlan, error) {
+	releasePlan := &appstudiov1alpha1.ReleasePlan{}
 
 	err := s.KubeRest().Get(context.TODO(), types.NamespacedName{
-		Name:      releaseName,
-		Namespace: releaseNamespace,
-	}, release)
+		Name:      releasePlanName,
+		Namespace: devNamespace,
+	}, releasePlan)
 
-	return release, err
+	return releasePlan, err
+}
+
+// DeleteReleasePlan deletes the releasePlan resource with the given name from the given namespace.
+// Optionally, it can avoid returning an error if the resource did not exist:
+//  specify 'false', if it's likely the ReleasePlan has already been deleted (for example, because the Namespace was deleted)
+func (s *SuiteController) DeleteReleasePlan(releasePlanName, devNamespace string, reportErrorOnNotFound bool) error {
+	releasePlan := appstudiov1alpha1.ReleasePlan{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      releasePlanName,
+			Namespace: devNamespace,
+		},
+	}
+	err := s.KubeRest().Delete(context.TODO(), &releasePlan)
+	if err != nil && !reportErrorOnNotFound && k8sErrors.IsNotFound(err) {
+		err = nil
+	}
+	return err
 }
 
-// Get ReleaseLink object from a given namespace
-func (s *SuiteController) GetReleaseLink(name string, namespace string) (*v1alpha1.ReleaseLink, error) {
-	namespacedName := types.NamespacedName{
-		Name:      name,
-		Namespace: namespace,
+// CreateReleasePlanAdmission creates a new ReleasePlanAdmission using the given parameters.
+func (s *SuiteController) CreateReleasePlanAdmission(name, source_namespace, application, target_namespace, AutoReleaseLabel string, releaseStrategy string) (*appstudiov1alpha1.ReleasePlanAdmission, error) {
```

Comment:
Everything in here is a string. Shouldn't need to write `string` twice, only once